### DISCUSSION TITLE
Ability to use app router like getInitialProps

### DISCUSSION THREAD

**Discussion by themre** [vercel/next.js#53240] [App Router]
Hi all,
I started exploring app router and the biggest win from pages is getInitialProps which runs page first time on the server and then if we navigate to the same page with next links, data is fetched on the client. Is there some way to achieve this thing same with app router? Because it seems it's always that page needs to served from the server or from the client.

**Comment by edsphinx** [comment]
I have the same curiosity to know exactly the same about how to use something similar to getInitialProps in app router, in the documentation suggest using fetch for data fetching but there isn't any use case in we can achieve the same behavior as getInitialProps

**Comment by kkx64** [comment]
Still no answer... this is a functionality we need

**Comment by revskill10** [comment]
Could you give typical use cases ?

**Reply by kkx64** [reply]
Im currently trying to load translations from an API. I can't find a good way to load them server-side and pass globally to client components without ultra-deep prop drilling. Before it would probably be achieved with getServerSideProps.

**Comment by rte76702** [comment]
I would also like to share my use case:

Currently, using getInitialProps, when a user **first** visits my site:
- page is rendered on the frontend server (FE) server, where I perform the following actions:
  - authentication using cookies
  - backend (BE) server token refresh if applicable
  - fetch data from the BE
  - render page on FE server
 
then on client-side navigation:
- display page-load animation
- check authentication
- check if data is available in cache, if not:
  - perform BE token refresh if applicable
  - fetch data from BE
- render page on browser

So on client-side navigation, the FE server is never involved, except when loading javascript chunks due to code splitting and dynamic imports.
This works quite well for me, but I am anxious about the possibility of getInitialProps no longer being supported/maintained.


I would love to find out if there is a way to implement my use-case with "App Router"
